cEP-0021.md

```diff
@@ -0,0 +1,349 @@
+# Profile Bears
+
+| Metadata |                                                   |
+| -------- | ------------------------------------------------- |
+| cEP      | 0021                                              |
+| Version  | 1.0                                               |
+| Title    | Debug and Profile Bears                           |
+| Authors  | Vaibhav Kumar Rai <mailto:raivaibhav08@gmail.com> |
+| Status   | Proposed                                          |
+| Type     | Process                                           |
+
+## Abstract
+
+This cEP describes the detailed implementation of profiling interface as a part
+of the
+[GSoC'18 project](https://summerofcode.withgoogle.com/projects/#6109762077327360).
+
+## Introduction
+
+Profiling a Python program is doing a dynamic analysis that measures the
+execution time of the program and everything that compose it. That means
+measuring the time spent in each of its functions. This will give the data about
+where our program is spending time, and what area might be worth optimizing,
+thus we can provide a *Profiling interface* from within coala and make it
+available to bear writers so that they can create Better Performant bears.
+
+This proposal is to introduce the Profiling interface so that bear writer will
+have the ability to Profile the Bear code to create Better Performant
+bears.
+
+## Proposed Change
+
+Bear writer can invoke the profiler through three ways:
```

Comment:
PEP 20:

> There should be one-- and preferably only one --obvious way to do it.

Reply:
So this should be:

"Bear writer can invoke the profiler with the `--profile-bears` argument"